modules/Hoot/ui/modals/changesetStats.js

```diff
@@ -0,0 +1,226 @@
+import FormFactory from '../../tools/formFactory';
+import { uiChangesetEditor } from '../../../ui/changeset_editor';
+
+export default class ChangesetStats {
+    constructor( job, data ) {
+        this.job = job;
+        this.changesetInfo = data;
+        this.includeTags = false;
+        this.changesetEditor = uiChangesetEditor(Hoot.context)
+            .on('change', changeTags);
+        var that = this;
+        function changeTags(changed, onInput) {
+            if (changed.hasOwnProperty('comment')) {
+                if (changed.comment === undefined) {
+                    changed.comment = '';
+                }
+                if (!onInput) {
+                    Hoot.context.storage('comment', changed.comment);
+                    Hoot.context.storage('commentDate', Date.now());
+                }
+            }
+            if (changed.hasOwnProperty('source')) {
+                if (changed.source === undefined) {
+                    Hoot.context.storage('source', null);
+                } else if (!onInput) {
+                    Hoot.context.storage('source', changed.source);
+                    Hoot.context.storage('commentDate', Date.now());
+                }
+            }
+            if (changed.hasOwnProperty('hashtags')) {
+                if (changed.hashtags === undefined) {
+                    Hoot.context.storage('hashtags', null);
+                } else if (!onInput) {
+                    Hoot.context.storage('hashtags', changed.hashtags);
+                    Hoot.context.storage('commentDate', Date.now());
+                }
+            }
+
+            that.tags = Object.assign(that.tags, changed);
+
+            that.changesetSection
+                .call(that.changesetEditor
+                    .tags(that.tags)
+                );
```

Comment:
this is the meat of the PR, adding the changeset editor used in commit.js to the apply changeset workflow:

![image](https://user-images.githubusercontent.com/2600932/70549542-6389fb00-1b42-11ea-957b-a644cc74045e.png)